src/internal/connector/exchange/exchange_data_collection.go

```diff
@@ -197,7 +197,21 @@ func (col *Collection) streamItems(ctx context.Context, errs *fault.Bus) {
 	}
 
 	// Limit the max number of active requests to GC
-	semaphoreCh := make(chan struct{}, urlPrefetchChannelBufferSize)
+	fetchParallelism := col.ctrl.ItemFetchParallelism
+	if fetchParallelism == 0 || fetchParallelism > urlPrefetchChannelBufferSize {
```

Comment:
Because you never know.
```suggestion
	if fetchParallelism < 1 || fetchParallelism > urlPrefetchChannelBufferSize {
```